diff --git a/firmware_symbols.h.S b/firmware_symbols.h.S
index bb91bca..11d1595 100644
--- a/firmware_symbols.h.S
+++ b/firmware_symbols.h.S
@@ -74,8 +74,6 @@
 
 #define RESIDENT_AREA_MAXSIZE   1452
 
-// Standalone payload, does not need file patching
-#define STANDALONE
 
 #elif defined(FIRMWARE_3_41)
 
@@ -133,7 +131,77 @@
        
 #define RESIDENT_AREA_MAXSIZE   (1296)
 
-#ifdef KIOSK
+#define HASH_TABLE_1            0xa0556f3d002cb8fd
+#define HASH_TABLE_2            0x6b70280200020017
+#define HASH_TABLE_3            0x8c0a948c000d99b1
+#define HASH_TABLE_4            0xa2bc1a5600052adc
+
+#define elf1_func1              0x5f3fc0
+#define elf1_func1_offset       0x00
+#define elf1_func2              0x305354
+#define elf1_func2_offset       0x14
+#define elf2_func1              0x2eb7c
+#define elf2_func1_offset       0x374
+
+#define elf3_data               0x0022b888
+#define elf4_data               0x000d68b8
+
+#elif defined (FIRMWARE_3_41_KIOSK)
+
+/* Common Symbols */
+#define memcpy                  0x7c01c
+#define memset                  0x4d144
+#define strcpy                  0x4d2f0
+#define strncmp                 0x4d344
+#define strlen                  0x4d318
+#define pathdup_from_user       0x1b3b3c // FIXME: unsure of the name!
+#define copy_from_user          0xE26C
+#define copy_to_user            0xE050
+#define alloc_and_copy_from_user 0x1b3CFC
+#define strdup_from_user        0x1B8A60
+#define alloc                   0x62088
+#define free                    0x624c8
+#define USBRegisterDriver       0xd22d8
+#define syscall_table           0x2eb128
+#define USBGetDeviceDescriptor  0xd2998 // FIXME: unsure of the name!
+#define USBOpenEndpoint         0xd29c4 // FIXME
+#define USBControlTransfer      0xd292c // FIXME
+#define memory_patch_func       0x4e81c
+#define patch_func1             0x0490AC
+#define patch_func1_offset      0x34
+#define patch_func2             0x04F07C
+#define patch_func2_offset      0x2C
+#define patch_func3             0x2aafc8
+#define patch_func3_offset      0x24
+#define patch_func4             0x04ed18
+#define patch_func4_offset      0x0
+#define patch_func5             0x0505d0
+#define patch_func5_offset      0x0
+#define patch_func6             0x0234d0
+#define patch_func6_offset      0x0
+#define patch_func7             0x0e83d4
+#define patch_func7_offset      0x0
+#define patch_func8             0x057214 //lv2open update patch
+#define patch_func8_offset1     0xA4 //lv2open update patch
+#define patch_func8_offset2     0x208 //lv2open update patch
+#define patch_func9             0x4F840 // must upgrade error
+#define patch_func9_offset      0x3EC
+#define patch_syscall_func      0x28FB34
+#define patch_data1             0x3ba890
+#define rtoc_entry_1            0xf08
+#define rtoc_entry_2            -0x6a00
+
+#define lv2_printf              0x28A654
+#define lv2_printf_null         0x28F050
+#define hvsc107_1               0xE5E8
+#define hvsc107_2               0xE67C
+#define hvsc107_3               0xE4B8
+
+// Payload bases
+#define MEM_BASE2               (0x50B3C)
+
+#define RESIDENT_AREA_MAXSIZE   (1296)
+
 #define HASH_TABLE_1            0xA0577BB4002C8B20
 #define HASH_TABLE_2            0x6B7028FE00020041
 #define HASH_TABLE_3            0x8c0a948c000d93D3
@@ -149,23 +217,71 @@
 #define elf3_data               0x0022A758
 #define elf4_data               0x000c86c8
 
-#else /* KIOSK */
+#elif defined (FIRMWARE_3_41_DEX)
 
-#define HASH_TABLE_1           0xa0556f3d002cb8fd
-#define HASH_TABLE_2           0x6b70280200020017
-#define HASH_TABLE_3           0x8c0a948c000d99b1
-#define HASH_TABLE_4           0xa2bc1a5600052adc
-
-#define elf1_func1             0x5f3fc0
-#define elf1_func1_offset      0x00
-#define elf1_func2             0x305354
-#define elf1_func2_offset      0x14
-#define elf2_func1             0x2eb7c
-#define elf2_func1_offset      0x374
-       
-#define elf3_data              0x0022b888
-#define elf4_data              0x000d68b8
-#endif /* KIOSK */
+/* Common Symbols */
+#define memcpy                  0x80014
+#define memset                  0x50A08
+#define strcpy                  0x50BB4
+#define strncmp                 0x50C08 //
+#define strlen                  0x50BDC
+#define pathdup_from_user       0x1B9EF8 // FIXME: unsure of the name!
+#define copy_from_user          0xE8B4 //
+#define copy_to_user            0xE698 //
+#define alloc                   0x65974 //
+#define free                    0x65DB4 //
+#define USBRegisterDriver       0xD7964 //
+#define syscall_table           0x303130
+#define USBGetDeviceDescriptor  0xD8024 // FIXME: unsure of the name! //
+#define USBOpenEndpoint         0xD8050 // FIXME
+#define USBControlTransfer      0xD7FB8 // FIXME //
+#define memory_patch_func       0x520E0
+#define patch_func1             0x4C970
+#define patch_func1_offset      0x34
+#define patch_func2             0x52940 //
+#define patch_func2_offset      0x2C
+#define patch_func3             0x2C07D0 //
+#define patch_func3_offset      0x24
+#define patch_func4             0x525DC
+#define patch_func4_offset      0x0
+#define patch_func5             0x053E94 //
+#define patch_func5_offset      0x0
+#define patch_func6             0x02509C
+#define patch_func6_offset      0x0
+#define patch_func7             0x0EDA60 //
+#define patch_func7_offset      0x0
+#define patch_func8             0x05AB00 //lv2open update patch //
+#define patch_func8_offset1     0xA4 //lv2open update patch
+#define patch_func8_offset2     0x208 //lv2open update patch
+#define patch_func9             0x53104 // must upgrade error
+#define patch_func9_offset      0x3EC
+#define patch_func10            0x5ACFC // 0x48000098 Debug retail package fix
+#define patch_syscall_func     0
+#define patch_data1             0x3D5B90
+#define rtoc_entry_1            0x1028
+#define rtoc_entry_2            -0x6920
+ 
+// Payload bases
+#define MEM_BASE2               (0x54400)
+        
+#define RESIDENT_AREA_MAXSIZE   (1296)
+ 
+#define HASH_TABLE_1            0
+#define HASH_TABLE_2            0
+#define HASH_TABLE_3            0
+#define HASH_TABLE_4            0
+ 
+#define elf1_func1              0
+#define elf1_func1_offset       0
+#define elf1_func2              0
+#define elf1_func2_offset       0
+#define elf2_func1              0
+#define elf2_func1_offset       0
+#define elf3_func1              0
+#define elf3_func2              0
+        
+#define elf3_data               0
+#define elf4_data               0
 
 #elif defined(FIRMWARE_3_40)
 
@@ -173,7 +289,7 @@
 #define memcpy                         0x7c018
 #define memset                         0x4d140
 #define strcpy                         0x4d2ec
-#define strncmp                        0x4d340
+#define strncmp                        0x4d340
 #define strlen                 0x4d314
 #define pathdup_from_user      0x1b3b34
 #define alloc                  0x62084

